src/mscorlib/shared/System/Threading/ReaderWriterLockSlim.cs

```diff
@@ -54,8 +53,10 @@ internal class ReaderWriterCount
     /// </summary>
     public class ReaderWriterLockSlim : IDisposable
     {
+        private static readonly int ProcessorCount = Environment.ProcessorCount;
```

Comment:
Assumes no processor affinity was set.
Is there any guideline regarding affinity? (ignore it? expect it?).